### PR TITLE
Sync upload button w/ progress

### PR DESCRIPTION
**Pull request by Chrischuck**
#534

I figured using an event emitter was probably the easiest path forward here.

https://user-images.githubusercontent.com/13311268/225186226-35db86ba-b097-41cf-b481-239ed00fb2ae.mov

### CODE REVIEW

**Review by albullington — CHANGES_REQUESTED** [review]
Neat, I like this approach. One thing I'm seeing is that the `cleanup( )` function in `useUploadObservations` clears the final status text that we want to display on the Toolbar (see screenshot below). I don't think we need that cleanup function when there are no more uploads -- instead, I added [a listener](https://github.com/inaturalist/iNaturalistReactNative/blob/c89fc837da5831f373696528b49f7bc3343de563/src/components/MyObservations/ToolbarContainer.js#L104) to the ToolbarContainer that should reset the Toolbar when a user leaves this screen.

But, I'm not picky about what approach we take as long as we display the correct status text to show a user that their uploads completed.

Screen Shot 2023-03-15 at 11 19 37 AM

**Comment by albullington** [comment]
The completed progress bar is still there in the design, so I don't think we need to reset it until after a user leaves the screen. Here's the link to the [Figma screens](https://www.figma.com/file/MChpvx4ZrKEVVwsWKt4lkI/iNaturalist-Mobile-UI-Design?node-id=2964%3A48949&t=zOjVkWzeRWtKTXLr-0) for what the toolbar should look like when an upload is completed.

**Comment by Chrischuck** [comment]
@albullington cool, reverted any changes on that side of things

File: src/providers/ObsEditProvider.js [hunk 1]
```diff
@@ -47,6 +49,25 @@ const ObsEditProvider = ( { children }: Props ): Node => {
     setUnsavedChanges( false );
   }, [] );
 
+  useEffect( () => {
+    const progressListener = EventRegister.addEventListener(
+      "INCREMENT_OBSERVATIONS_PROGRESS",
+      increments => {
+        setUploadProgress( currentProgress => {
+          increments.forEach( ( [uuid, increment] ) => {
+            currentProgress[uuid] ??= 0;
```
Comment (albullington) [inline src/providers/ObsEditProvider.js:58]: @Chrischuck what is `currentProgress[uuid] ??= 0;` doing here? The `??=` operation seems to be causing e2e tests to fail, and I'm not sure why this line is necessary.

**Pull request merged**